src/jshint.js

```diff
@@ -3374,9 +3380,27 @@ var JSHINT = (function() {
         advance(")");
       } else {
         var is_rest = checkPunctuator(state.tokens.next, "...");
-        ident = identifier();
-        if (ident)
+
+        if (isAssignment) {
+          var identifierToken = is_rest ? peek(0) : state.tokens.next;
+          if (!identifierToken.identifier) {
+            warning("E030", identifierToken, identifierToken.value);
+          }
+          var assignTarget = expression(155);
+          if (assignTarget) {
+            checkLeftSideAssign(assignTarget);
+
+            // if the target was a simple identifier, add it to the list to return
+            if (assignTarget.identifier) {
```

Comment:
The converse of this condition is not currently tested. In other words, this also passes:

```diff
-// if the target was a simple identifier, add it to the list to return
-if (assignTarget.identifier) {
   ident = assignTarget.value;
-}
```

...but that's clearly wrong. I'm having trouble thinking of a test case for that, though. Any ideas?